### ISSUE TITLE
shadow of Dropdowns showing up when used on scrolling container

### ISSUE THREAD

**Issue by lglokeshgandham** [MyreMylar/pygame_gui#153]
**Describe the bug**
 dropdowns are visible outside the window when  scrolling container is used for window
**To Reproduce**
Steps to reproduce the behaviour:
1. Make a scrollable window.
2. Initialize a dropdown outside the window range but within the range of scrollable container.
3.shadow of dropdown is visible.

**Expected behaviour**
Dropdown shadow shouldn't be visible when initializing the window.
currently dropdown shadow(in dropdown1 image) becomes invisible whenever I click on window/Window scroll bars
**Screenshots**
![dropdown2](https://user-images.githubusercontent.com/44719011/88897022-48523400-d268-11ea-8c10-52d45c3bc6a2.png)
![dropdown1](https://user-images.githubusercontent.com/44719011/88897029-4a1bf780-d268-11ea-8062-b5f459770702.png)
Adding the code for above.

[Test.txt](https://github.com/MyreMylar/pygame_gui/files/4999819/Test.txt)

**Platform and software (please complete the following information):**
 - OS: [e.g. Windows]
 - Pygame GUI version [0.5.7]
 - Pygame version [e.g. 1.9.6]

**Additional context**
The above problem does not exist in version 0.5.6 .

**Comment by MyreMylar** [comment]
Thanks for the report. 

Looks like something being missed when set_image is called with updated clipping info.

Think I need to add a ScrollableContainer to the general test project to help catch visual things like this in the future..

**Comment by MyreMylar** [comment]
Managed to reproduce, and then find a fix for this bug, fix will be in 0.6.0. The secret was to always set your element's image attribute with `set_image()`. I also noticed that rebuild was being called twice when first creating a drop down so I fixed that as well.

**Test case used:**
```
import pygame
import pygame_gui

from pygame_gui.elements.ui_scrolling_container import UIScrollingContainer


class TestWindow(pygame_gui.elements.UIWindow):
    def __init__(self, rect, manager, scrollable_size):
        super().__init__(rect, manager,
                         window_display_title="Test Window",
                         object_id='#test_window',
                         resizable=True)

        container_rect = pygame.Rect((0, 0),
                                     self.get_container().get_size())
        self.scrolling_container = UIScrollingContainer(relative_rect=container_rect,
                                                        manager=self.ui_manager,
                                                        container=self,
                                                        anchors={'left': 'left',
                                                                 'right': 'right',
                                                                 'top': 'top',
                                                                 'bottom': 'bottom'})

        self.scrolling_container.set_scrollable_area_dimensions(scrollable_size)

        self.drop_down = pygame_gui.elements.UIDropDownMenu(['Apples', 'Pears', 'Oranges'],
                                                            'Apples',
                                                            pygame.Rect((450, 20), (100, 30)),
                                                            manager=manager,
                                                            container=self.scrolling_container)


pygame.init()


window_surf = pygame.display.set_mode((1200, 800))
window_surf.fill((20, 20, 40))

manager = pygame_gui.ui_manager.UIManager((1200, 800))

scrolling_container = TestWindow(pygame.Rect((10, 10), (400, 300)), manager=manager,
                                 scrollable_size=(600, 600))


clock = pygame.time.Clock()
running = True

while running:
    time_delta = clock.tick(60)
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

        manager.process_events(event)

    manager.update(time_delta)
    window_surf.fill((20, 20, 40))
    manager.draw_ui(window_surface=window_surf)

    pygame.display.update()

```
